### ISSUE TITLE
Issue building Cufflinks from source

### ISSUE THREAD

**Issue by shashidhar22** [cole-trapnell-lab/cufflinks#68]
Configure is not able to recognize eigen , even after I export the path to eigen:

``` sh
export EIGEN3_CPPFLAGS="-I/projects/home/sravishankar9/local/include/Eigen"
```

When configure is run with the following command : 

``` sh
./configure --prefix=/projects/home/sravishankar9/local --with-boost=/projects/home/sravishankar9/local
```

The following warning is observed:

``` sh
./configure: line 7616: AX_EIGEN: command not found
```

Getting the following error on make:

``` sh
GHash.hh: In member function ‘GHash<OBJ>::GHashEntry* GHash<OBJ>::NextEntry()’:
GHash.hh:91:44: error: type/value mismatch at argument 1 in template parameter list for ‘template<class _Tp> struct std::hash’
     while (pos<fCapacity && hash[pos].hash<0) pos++;
                                            ^
GHash.hh:91:44: note:   expected a type, got ‘0’
GHash.hh: In member function ‘char* GHash<OBJ>::NextKey()’:
GHash.hh:417:41: error: type/value mismatch at argument 1 in template parameter list for ‘template<class _Tp> struct std::hash’
  while (pos<fCapacity && hash[pos].hash<0) pos++;
                                         ^
GHash.hh:417:41: note:   expected a type, got ‘0’
GHash.hh: In member function ‘OBJ* GHash<OBJ>::NextData()’:
GHash.hh:430:41: error: type/value mismatch at argument 1 in template parameter list for ‘template<class _Tp> struct std::hash’
  while (pos<fCapacity && hash[pos].hash<0) pos++;
                                         ^
GHash.hh:430:41: note:   expected a type, got ‘0’
GHash.hh: In member function ‘OBJ* GHash<OBJ>::NextData(char*&)’:
GHash.hh:444:41: error: type/value mismatch at argument 1 in template parameter list for ‘template<class _Tp> struct std::hash’
  while (pos<fCapacity && hash[pos].hash<0) pos++;
                                         ^
GHash.hh:444:41: note:   expected a type, got ‘0’
make[2]: *** [clustering.o] Error 1
make[2]: Leaving directory `/projects/home/sravishankar9/software/cufflinks/src'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/projects/home/sravishankar9/software/cufflinks'
make: *** [all] Error 2
```

Specfications of system and pre-requisites installed:

Distro : Red Hat Enterprise Linux Server release 6.7 (Santiago)
GCC : gcc version 7.0.0 20160701 (experimental) (GCC)
Boost :Version 1.55
Eigen : Version 3.2.8
Samtools : Version 1.3.1
- All the instructions mentioned in the README have been followed. 

Any help in resolving this issue would be appreciated.
Thank you

**Comment by gpertea** [comment]
The make errors shown here don't seem to have anything to do with Eigen detection, but rather to a custom string hash template (of which I am partially guilty) implemented in GHash.hh which apparently is disliked by the compiler you [attempted to] use.
      It baffles me how someone would choose to compile years-old C++ scientific code with a not yet released, **experimental** C++ compiler (which is not even in beta stage perhaps ?) and then wonders why the code does not compile properly.. And ironically this is attempted on a years-old platform (RHEL 6) which has absolutely no support for newer GCC compilers like 4.9 and beyond.

Could you perhaps try to first compile the code with an officially released and supported C++ compiler and let us know if you encounter any errors?

**Comment by jmarshall** [comment]
> ./configure: line 7616: AX_EIGEN: command not found

This is because the relevant `m4_include` was still missing — can be fixed by applying #76.

**Comment by mestia** [comment]
Hi All, I am having the same issues with gcc-7, with the difference that GCC-7 is already stable but the old scientific code is still old :) Any hint how to fix the issues are greatly appreciated. The system where it fails are Debian testing and sid.

**Comment by gpertea** [comment]
See issue #89, this was actually fixed in GCC 7.1.1, so it looks like this was a rather persistent parser bug which made it into the earlier (stable?) versions of GCC 7..
